zerver/webhooks/github/headers.py

```diff
@@ -0,0 +1,55 @@
+pull_request_headers = {"HTTP_X_GITHUB_EVENT": "pull_request"}
+
+fixtures_to_headers_map = {
+    "assigned_pull_request": pull_request_headers,
+    "check_run_completed": {"HTTP_X_GITHUB_EVENT": "check_run"},
+    "check_run_in_progress": {"HTTP_X_GITHUB_EVENT": "check_run"},
+    "closed_pull_request": pull_request_headers,
+    "commit_comment": {"HTTP_X_GITHUB_EVENT": "commit_comment"},
+    "create": {"HTTP_X_GITHUB_EVENT": "create"},
+    "delete": {"HTTP_X_GITHUB_EVENT": "delete"},
+    "deployment": {"HTTP_X_GITHUB_EVENT": "deployment"},
+    "deployment_status": {"HTTP_X_GITHUB_EVENT": "deployment_status"},
+    "edited_pull_request": pull_request_headers,
+    "fork": {"HTTP_X_GITHUB_EVENT": "fork"},
+    "issue": {"HTTP_X_GITHUB_EVENT": "issues"},
+    "issue_comment": {"HTTP_X_GITHUB_EVENT": "issue_comment"},
+    "issue_comment_deleted": {"HTTP_X_GITHUB_EVENT": "issue_comment"},
+    "labeled_pull_request": pull_request_headers,
+    "member": {"HTTP_X_GITHUB_EVENT": "member"},
+    "membership": {"HTTP_X_GITHUB_EVENT": "membership"},
+    "membership_removal": {"HTTP_X_GITHUB_EVENT": "membership"},
+    "merged_pull_request": pull_request_headers,
+    "opened_pull_request": pull_request_headers,
+    "opened_pull_request_with_preassigned_assignee": pull_request_headers,
+    "page_build": {"HTTP_X_GITHUB_EVENT": "page_build"},
+    "ping": {"HTTP_X_GITHUB_EVENT": "ping"},
+    "ping_organization": {"HTTP_X_GITHUB_EVENT": "ping"},
+    "public": {"HTTP_X_GITHUB_EVENT": "public"},
+    "pull_request_review": {"HTTP_X_GITHUB_EVENT": "pull_request_review"},
+    "pull_request_review_comment": {"HTTP_X_GITHUB_EVENT": "pull_request_review_comment"},
+    "pull_request_review_requested": pull_request_headers,
+    "pull_request_review_requested_multiple_reviewers": pull_request_headers,
+    "pull_request_review_requested_singular_key": pull_request_headers,
+    "push_1_commit": push_headers,
+    "push_1_commit_without_username": push_headers,
+    "push_50_commits": push_headers,
+    "push_delete_branch": push_headers,
+    "push_local_branch_without_commits": push_headers,
+    "push_multiple_committers": push_headers,
+    "push_multiple_committers_with_others": push_headers,
+    "push_tag": push_headers,
+    "release": {"HTTP_X_GITHUB_EVENT": "release"},
+    "repository": {"HTTP_X_GITHUB_EVENT": "repository"},
+    "repository_vulnerability_alert": {"HTTP_X_GITHUB_EVENT": "repository_vulnerability_alert"},
+    "request_review_removed_pull_request": pull_request_headers,
+    "status": {"HTTP_X_GITHUB_EVENT": "status"},
+    "status_with_target_url": {"HTTP_X_GITHUB_EVENT": "status"},
+    "synchronized_pull_request": pull_request_headers,
+    "team_add": {"HTTP_X_GITHUB_EVENT": "team_add"},
+    "unassigned_pull_request": pull_request_headers,
+    "unlabeled_pull_request": pull_request_headers,
+    "watch_repository": {"HTTP_X_GITHUB_EVENT": "watch"},
+    "wiki_pages": {"HTTP_X_GITHUB_EVENT": "gollum"},
+}
```

Comment:
Okay, just to reiterate, my point is that this takes the work from one module (`tests.py`) and simply moves it to a new module (`headers.py`) that the developer now has to worry about. It is just much simpler to say "hey pass in custom headers to `send_and_test_stream_message`, if you have any, otherwise don't worry about it". :)

Reply:
Yeah, agreed, this mapping file isn't super clean.  See my comment above -- there's a clean way to just eliminate this work.

Reply:
Right, changing the system from having `fixture_filename_to_headers` be a method to having it be a dictionary could really start to show the benefits of adding this small layer of extra (optional) complexity.